File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.py
from collections import defaultdict
def solution(k, tangerine):
    num = defaultdict(int)
    for size in tangerine:
        num[size-1] += 1
    
    m = len(tangerine) - k
    
    result = []
    for item in num:
        if(num[item] <= m):
            result.append(num[item])
    
    result.sort(reverse=True)
    
    cnt = 0
    while(m != 0):
        if(result == []):
            break
        cur = result.pop()
        if(m >= cur):
            m -= cur
            cnt +=1
            
    return len(num) - cnt

File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        int n = tangerine.length;
        
        HashMap<Integer, Integer> num = new HashMap<Integer, Integer>();
        
        for(int i=0;i<n;i++){
            if(num.get(tangerine[i]) == null){
                num.put(tangerine[i], 1);    
            }
            else{
                num.put(tangerine[i], num.get(tangerine[i]) + 1);
            }
        }
        
        ArrayList<Integer> numberOfSize = new ArrayList<>();
        
        for(int i : num.keySet()){
            numberOfSize.add(num.get(i));
        }
        
        int pick = 0;
        int answer = 0;
        
        Collections.sort(numberOfSize, Collections.reverseOrder());
        while(pick < k){
            pick += numberOfSize.remove(0);
            answer += 1;
            if(pick >= k){
                break;
            }
            
        }
        return answer;
    }
}